src/sys/unix/linux.rs

```diff
@@ -0,0 +1,32 @@
+use std::{convert::TryInto, fs, io, mem::size_of, os::unix::io::AsRawFd, path::Path};
+
+use ioctl_sys::iow;
+use libc::c_int;
+
+use super::super::utility::NamedTempFile;
+
+const C_INT_SIZE: usize = size_of::<c_int>();
+const FICLONE: u32 = iow!(0x94, 9, C_INT_SIZE);
+
+pub fn reflink(from: &Path, to: &Path) -> io::Result<()> {
+    let src = fs::File::open(from)?;
+
+    // pass O_EXCL to mimic macos behaviour
+    let dest = NamedTempFile::create_new(to)?;
```

Comment:
Why use a temporary file when the original doesn't?

Reply:
Its name is a bit confusing, it is actually just an RAII that automatically remove the file on drop.

Reply:
I renamed it to `AutoRemovedFile`.

Reply:
So, this is not actually a temp file, it's a file that deletes itself on drop unless persisted?

Reply:
> So, this is not actually a temp file, it's a file that deletes itself on drop unless persisted?

Yep